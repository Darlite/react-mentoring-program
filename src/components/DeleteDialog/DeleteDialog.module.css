.deleteMessage {
    font-size: 20px;
}

.confirmDeleteMovieButton {
    background: #F65261;
    text-transform: uppercase;
    font-size: 20px;
    font-weight: 500;
    color: #FFFFFF;
    border: none;
    padding: 10px 35px;
    align-self: flex-end;
    cursor: pointer;
}
