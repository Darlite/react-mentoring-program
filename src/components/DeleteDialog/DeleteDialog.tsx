import React from "react";
import styles from "../DeleteDialog/DeleteDialog.module.css";
import {MovieDetailsData} from "../../types/MovieDetailsData";

interface DeleteDialogProps {
    movieToDelete: MovieDetailsData | null,
}

const DeleteDialog: React.FC<DeleteDialogProps> = ({ movieToDelete }) => {
    return (
        <>
            <p className={styles.deleteMessage}>
                Are you sure you want to delete
                <span className={styles.movieToDeleteTitle}> {movieToDelete?.title} </span>
                movie?
            </p>
            <button className={styles.confirmDeleteMovieButton}
                    aria-label="Confirm"
            >
                Confirm
            </button>
        </>
    )
}

export default DeleteDialog;
