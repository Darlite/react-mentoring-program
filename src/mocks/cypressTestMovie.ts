import {GenreType} from "../constants/GenreType";

export const cypressTestMovie = {
    title: "Cypress Test Movie",
    editedMovieTitle: "Cypress Edited Test Movie",
    release_date: "2025-05-04",
    poster_path: "https://www.movieposters.com/cdn/shop/files/tron-ares_avxg7mih_480x.progressive.jpg",
    genres: [GenreType.Action, GenreType.ScienceFiction],
    runtime: "138",
    vote_average: "7.2",
    overview: "A highly sophisticated Program, Ares, is sent from the digital world into the real world on a dangerous mission.",
}
