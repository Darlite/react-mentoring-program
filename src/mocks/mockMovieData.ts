import {GenreType} from "../constants/GenreType";

export const moviesData = {
    id: 1,
    imageUrl: "images/Pulp_Fiction.png",
    title: "Pulp Fiction",
    releaseDate: "1994-09-23",
    movieUrl: "https://www.imdb.com/title/tt0110912",
    genres: [GenreType.Crime, GenreType.Thriller],
    runtime: "2h 34m",
    rating: "8.9",
    description: "Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra",
}
