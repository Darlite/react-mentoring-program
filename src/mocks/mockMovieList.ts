import {GenreType} from "../constants/GenreType";

export const mockMovieList = [
    {
        id: 1,
        poster_path: "images/Pulp_Fiction.png",
        title: "Pulp Fiction",
        release_date: "1994-09-23",
        movieUrl: "https://www.imdb.com/title/tt0110912",
        genres: [GenreType.Crime, GenreType.Thriller],
        runtime: "2h 34m",
        vote_average: "8.9",
        overview: "Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra",
    },
    {
        id: 2,
        poster_path: "images/Bohemian_Rhapsody.png",
        title: "Bohemian Rhapsody",
        release_date: "2018-10-30",
        movieUrl: "https://www.imdb.com/title/tt1727824",
        genres: [GenreType.Musical, GenreType.Documentary],
        runtime: "2h 15m",
        vote_average: "8.5",
        overview: "With his impeccable vocal abilities, Freddie Mercury and his rock band, Queen, achieve superstardom. However, amidst his skyrocketing success, he grapples with his ego, sexuality and a fatal illness.",
    },
    {
        id: 3,
        poster_path: "images/Kill_Bill_Volume_2.png",
        title: "Kill Bill: Vol. 2",
        release_date: "2004-04-08",
        movieUrl: "https://www.imdb.com/title/tt0378194",
        genres: [GenreType.Action, GenreType.Thriller],
        runtime: "2h 17m",
        vote_average: "8.9",
        overview: "A pregnant woman, codenamed the Bride, sets out on a journey to kill her ex-boss, Bill, and targets his brother, Budd, and Elle Driver, the only two survivors of the Deadly Vipers Assassination Squad.",
    },
]