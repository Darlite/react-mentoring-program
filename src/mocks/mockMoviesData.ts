import {GenreType} from "../constants/GenreType";

export const mockMoviesData = {
    id: 680,
    poster_path: "https://www.movieposters.com/cdn/shop/files/PulpFiction.Ver1_480x.progressive.jpg?v=1741192018",
    title: "Pulp Fiction",
    release_date: "1994-09-10",
    genres: [GenreType.Crime, GenreType.Thriller],
    runtime: "167",
    vote_average: "8.3",
    overview: "Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra",
}
