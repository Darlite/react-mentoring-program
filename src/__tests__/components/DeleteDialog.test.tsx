import React from "react";
import {render} from "@testing-library/react";
import DeleteDialog from "../../components/DeleteDialog/DeleteDialog";
import {mockMoviesData} from "../../mocks/mockMoviesData";

describe("DeleteDialog", () => {
    it("renders correctly", () => {
        const { container } = render(<DeleteDialog movieToDelete={mockMoviesData}/>);
        expect(container).toMatchSnapshot();
    });
})