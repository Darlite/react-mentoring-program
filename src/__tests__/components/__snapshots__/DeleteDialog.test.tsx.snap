// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteDialog renders correctly 1`] = `
<div>
  <p
    class="deleteMessage"
  >
    Are you sure you want to delete
    <span
      class="movieToDeleteTitle"
    >
       
      Pulp Fiction
       
    </span>
    movie?
  </p>
  <button
    aria-label="Confirm"
    class="confirmDeleteMovieButton"
  >
    Confirm
  </button>
</div>
`;
